components/operator/internal/resources/payments/deployments.go

```diff
@@ -298,3 +375,19 @@ func createGateway(ctx core.Context, stack *v1beta1.Stack, p *v1beta1.Payments)
 		New(p, deploymentTemplate).
 		Install(ctx)
 }
+
+func validateTemporalURI(temporalURI *v1beta1.URI) error {
+	if temporalURI.Scheme != "temporal" {
+		return fmt.Errorf("invalid temporal uri: %s", temporalURI.String())
+	}
+
+	if temporalURI.Path == "" {
+		return fmt.Errorf("invalid temporal uri: %s", temporalURI.String())
+	}
+
+	if !strings.HasPrefix(temporalURI.Path, "/") {
+		return fmt.Errorf("invalid temporal uri: %s", temporalURI.String())
+	}
```

Comment:
consider using the net/url package to normalise the url so that both with and without leading slash is accepted:

```suggestion
	u, err := url.JoinPath(temporalURI.Path, "/")
	if err != nil {
		return fmt.Errorf("invalid temporal uri: %s", temporalURI.String())
	}
	temporalURI = u
```